### PR TITLE
feat(fedimint-cli): create fedimint-cli module list subcommand

### PR DESCRIPTION
**Pull request by 15IITian**
* Created `ModuleStatus` struct containing status of a module, `ModuleSubcommand `enum containing `List` variant
 * Created `Data` struct which contains module related data
 * On running `fedimint-cli module list` command -> will store all federation supported modules in a Vector of `Data` Struct.
 
 #3119

### CODE REVIEW

**Comment by 15IITian** [comment]
image

To access client-supported modules -> I try to take `modules` field from the `client` variable -> as  `modules: ClientModuleRegistry` which has a function `iter-modules`  to  get all modules.

But `Client` struct fields are `private` and I  found no such function in its implementation to iterate over modules .

**Comment by 15IITian** [comment]
as per https://github.com/fedimint/fedimint/issues/3119#issuecomment-1913091136 -> I have not included `recovering`  for now.

File: fedimint-cli/src/client.rs [hunk 1]
```diff
@@ -161,15 +161,25 @@ pub enum ClientCmd {
     },
     /// Call a module subcommand
     Module {
+        #[clap(subcommand)]
+        command: ModuleSubcommand,
         /// Module selector (either module id or module kind)
-        #[clap(long)]
-        module: ModuleSelector,
+        module: Option<ModuleSelector>,
         args: Vec<ffi::OsString>,
```
Comment (Maan2003) [inline fedimint-cli/src/client.rs:168]: likely subcommand and args are incompatible

Comment (15IITian) [reply fedimint-cli/src/client.rs:168]: is it because of making it a postional argument ?

Comment (Maan2003) [reply fedimint-cli/src/client.rs:168]: no, just incompatible
args should be part of subcommand

Comment (15IITian) [reply fedimint-cli/src/client.rs]: @Maan2003 what can be the problem now ?

Comment (15IITian) [reply fedimint-cli/src/client.rs]: I think,  backward incompatibility test fails due to this error  ->

image


Please tell how to remove it.

**Comment by justinmoon** [comment]
No progress in 3 weeks. Closing.

**Pull request closed without merging**